stages/org.osbuild.dnf.config

```diff
@@ -61,12 +79,56 @@ def configure_variable(tree, name, value):
         f.write(value + "\n")
 
 
+def make_value(value):
+    val = str(value)
+    return val
+
+
+def make_section(cfg, name, settings):
+    if not cfg.has_section(name):
+        cfg.add_section(name)
+
+    for key, value in settings.items():
+        val = make_value(value)
+        cfg.set(name, key, val)
+
+
+def make_dnf_config(tree, config_options):
+    """
+    Merges the given config object into /etc/dnf/dnf.conf, overwriting existing
+    values.
+    """
+    dnf_config_path = f"{tree}/etc/dnf/dnf.conf"
+    dnf_config = iniparse.SafeConfigParser()
+
+    try:
+        with open(dnf_config_path, "r") as f:
+            dnf_config.readfp(f)
+    except FileNotFoundError:
+        print(f"Warning: DNF configuration file '{dnf_config_path}' does not exist, will create it.")
+        dnf_dir = f"{tree}/etc/dnf"
+        if not os.path.isdir(dnf_dir):
+            assert not os.path.exists(dnf_dir)
+            os.mkdir(dnf_dir)
```

Comment:
Shorter: `os.makedirs(dnf_dirs, exist_ok=True)`. I think this is fine in our case sine it is a private tree that only we write to, but I am always weary of TOCTOU races ;)

Reply:
will that error out if /etc/dnf exists but isn't a directory? (probably a non-issue)

Reply:
ah right, private tree, nvm; fixing :)

Reply:
It is using `mkdir` under the hood[1] which I guess would give `ENOTDIR` (hopefully). Looking at the source it also does a `path.exists` check so it is not actually better than your implementation, _sigh_. 

[1] https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/os.py#L200

Reply:
eh, whatever you prefer then :D

Reply:
`os.makedirs` is shorter and used all over the source. And you have made the change, so lets just keep it that way! ;)